api/v1/calico_node_types.go

```diff
@@ -77,6 +77,12 @@ type CalicoNodeDaemonSetPodSpec struct {
 	// +optional
 	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
 
+	// TopologySpreadConstraints describes how a group of pods ought to spread across topology
+	// domains. Scheduler will schedule pods in a way which abides by the constraints.
+	// All topologySpreadConstraints are ANDed.
+	// +optional
+	TopologySpreadConstraints []v1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
```

Comment:
Does this make sense for a daemonset?

Reply:
It's certainly allowed for a DaemonSet in the API, although I agree the use-cases for a DaemonSet (and specifically calico/node) are less clear.